### PR TITLE
#159951822 Users should be able to create articles

### PR DESCRIPTION
**Pull request by RutaleIvanPaul**
#### What does this PR do?
Adds functionality to enable users to create articles.
#### What are the relevant pivotal tracker stories?
[#159951822](https://www.pivotaltracker.com/story/show/159951822)
#### Coverage Screenshot
screen shot 2018-09-11 at 17 05 36

### CODE REVIEW

**Comment by abulojoshua1** [comment]
@RutaleIvanPaul, why do we have two navigation bars on the application while on the home page?
screen shot 2018-09-10 at 4 40 05 pm

File: .eslintrc.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "react/jsx-sort-props": 0,
     "react/jsx-uses-react": 1,
     "react/jsx-uses-vars": 1,
-    "react/no-danger": 1,
```
Comment (lytes20) [inline .eslintrc.json]: Why did you disable this warning?

File: .eslintrc.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
...
+    "react/no-danger": 0,
```
Comment (lytes20) [inline .eslintrc.json]: @RutaleIvanPaul  why did you disable this warning?

File: src/setupTests.js [hunk 3]
```diff
@@ -5,12 +5,11 @@ configure({ adapter: new Adapter() });
...
 class LocalStorageMock {
   constructor() {
-    this.store = {
-    };
+    this.store = {};
   }
 
   getItem(key) {
-    return this.store[key] || false;
+    return this.store[key] || null;
```
Comment (lytes20) [inline src/setupTests.js:12]: @RutaleIvanPaul why did you change this fallback for `getItem` to null? Don't you think `false` is more workable with?

File: src/CSS/login.css [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    background: red; 
+    display: table;
+    color: #fff;
+     }
+
+
+
+/* input[type="file"] {
+    display: none;
+} */
```
Comment (abulojoshua1) [inline src/CSS/login.css]: @RutaleIvanPaul, please take out the commented out code if it is not necessary.

File: src/CSS/login.css [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+.new_data{
+    border: none;
+    border-bottom: 1px solid #000;
+    border-radius: 0;
+    padding-left: 0;
+    padding-right: 0;
+}
+
+.fr-toolbar.fr-top{
+    /* display: none; */
```
Comment (abulojoshua1) [inline src/CSS/login.css]: @RutaleIvanPaul, Why does this CSS class empty?

**Comment by lytes20** [comment]
@RutaleIvanPaul, How can I test this feature manually?

**Review by abulojoshua1 — CHANGES_REQUESTED** [review]

**Comment by abulojoshua1** [comment]
@RutaleIvanPaul, the drop-down on the profile link does not direct to your route, please fix that.

File: src/CSS/login.css (hunk 5 above)
Comment (lytes20) [reply src/CSS/login.css]: It has commented out code that you are not using, so you could just delete the class

**Comment by abulojoshua1** [comment]
@RutaleIvanPaul, your `/createArticle` route is not protected, I can view it even when I am not logged in, I think you should re-direct non-logged in users to the homepage route  i.e. `/`

File: src/setupTests.js (hunk 3 above)
Comment (abulojoshua1) [reply src/setupTests.js:12]: @lytes20, I think that it still does the same thing.

Comment (RutaleIvanPaul) [reply src/setupTests.js:12]: @lytes20 , It does the same thing and I have changed @abulojoshua1 's file that needs it to conform. Mine is also more semantically correct, since you can either have a value or a null.

File: src/CSS/login.css (hunk 5 above)
Comment (RutaleIvanPaul) [reply src/CSS/login.css]: I have changed this.

**Comment by RutaleIvanPaul** [comment]
@lytes20  Check it out on heroku

**Review by abulojoshua1 — APPROVED** [review]

**Review by ssewilliam — COMMENTED** [review]
@RutaleIvanPaul,
- Is there a specific reason why you have named the file 
 createArticle as .jsx and not .js

**Comment by ssewilliam** [comment]
@RutaleIvanPaul,
- This PR is failing on Heroku
screen shot 2018-09-11 at 10 35 29

**Comment by AGMETEOR** [comment]
@RutaleIvanPaul the actions "START_LOGIN" AND "START_REGISTRATION" were changed to "OPEN_MODAL_TO_LOGIN " and "OPEN_MODAL_TO_REGISTRATION".

**Comment by AGMETEOR** [comment]
@RutaleIvanPaul I have just been trying out the feature and it's great. However, when I upload an image I get no visual feedback of what's going on. I am not sure whether my image was uploaded successfully or not.

**Review by AGMETEOR — APPROVED** [review]

**Comment by RutaleIvanPaul** [comment]
@AGMETEOR , I have rectified that

**Review by ssewilliam — APPROVED** [review]

File: src/components/Articles/createArticle.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+
+  eventListener = event => this.setState({ [event.target.name]: event.target.value });
+
+  uploadImage = (picture) => {
+    this.setState({ publishButtonDisabled: true });
+    this.setState({ imageUploadState: `fa fa-2x fa-spinner fa-spin ${classes.fa}` });
+    const formdata = new FormData();
+    formdata.append('file', picture);
+    const API_KEY = '826821331683246';
```
Comment (geofrocker) [inline src/components/Articles/createArticle.js]: Store these keys as environment variables
